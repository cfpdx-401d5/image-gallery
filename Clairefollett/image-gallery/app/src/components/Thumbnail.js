import React from 'react';

export default function Thumbnail(props) {
    const thumbnail = props.puppies.map(puppies => {
        return (
            <ThumbnailDisplay key={puppies._id} puppies={puppies} onDelete={props.onDelete}/>
        );
    });
    return (
      <ul>{thumbnail}</ul>
    );
}

function ThumbnailDisplay(props) {
  return (
    <div className="clearfix">
      <li>
        <p height='20px' width='200'>title: {props.puppies.title}</p>
        <img className='thumbnail' src={props.puppies.url} alt='' height='100' width='100' />
        <div className="delete"><button className="shadow animate orange" onClick={() => props.onDelete(props.puppies._id)}>Delete</button></div>
      </li>
    </div>
  );
}
