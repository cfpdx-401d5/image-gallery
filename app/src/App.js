import React, { Component } from 'react';

import ListDisplay from './displays/ListDisplay';
import ThumbnailDisplay from './displays/ThumbnailDisplay';
import GalleryDisplay from './displays/GalleryDisplay';
import DisplaySelector from './DisplaySelector';
import FormContainer from './image_form/FormContainer';

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      displayType: 'list',
      images: [{
        id: 'id1',
        title: 'Cute Bunny',
        description: 'Isn\'t it fuzzy-wuzzy cutest thing you\'ve ever seen?',
        url: 'http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg'
      },
      {
        id: 'id2',
        title: 'Upside Down Sloth',
        description: 'Sloths just love to hang out.',
        url: 'http://kids.nationalgeographic.com/content/dam/kids/photos/animals/Mammals/Q-Z/sloth-beach-upside-down.jpg.adapt.945.1.jpg'
      },
      {
        id: 'id3',
        title: 'Waving Panda',
        description: 'When a panda waves, you wave back!',
        url: 'http://media4.s-nbcnews.com/j/newscms/2016_36/1685951/ss-160826-twip-05_8cf6d4cb83758449fd400c7c3d71aa1f.nbcnews-ux-2880-1000.jpg'
      }
      ],
    };
    this.setDisplay = this.setDisplay.bind(this);
    this.addImage = this.addImage.bind(this);
    this.deleteImage = this.deleteImage.bind(this);
  }

  setDisplay(whatDisplay) {
    this.setState({ displayType: whatDisplay });
  }

  addImage(e, imageSubmission) {
    e.preventDefault();
    // let newImgArray = [...this.state.images]; // copy the images[] in the state
    // newImgArray.push(imageSubmission); // make new image array

    // this.setState({ images: newImgArray });
    fetch('http://localhost:3001/images', {
      method: 'POST',
      mode: 'cors',
      headers: {
        'Accept': 'application/json',
        'Content-type': 'application/json'
      },
      body: JSON.stringify(imageSubmission)
    })
      .then(res => res.json())
      .then(res => {
        let newImgArray = [...this.state.images];
        delete res.__v;
        newImgArray.push(res);
        this.setState({ images: newImgArray });
      });
  }

  deleteImage(imageID) {
    // let newImgArray = this.state.images.filter(image => {
    //   return (image.id !== imageID);
    // });

    // this.setState({ images: newImgArray });
    fetch(`http://localhost:3001/images/${imageID}`, { method: 'DELETE' })
      .then(res => res.json())
      .then(res => {
        if (res.message) {
          let newImgArray = this.state.images.filter(image => {
            return (image.id !== imageID);
          });
          this.setState({ images: newImgArray });
        }
      });
  }

  render() {
    let DisplayType;

    if (this.state.displayType === 'list') DisplayType = ListDisplay;
    else if (this.state.displayType === 'thumbnail') DisplayType = ThumbnailDisplay;
    else if (this.state.displayType === 'gallery') DisplayType = GalleryDisplay;

    return (
        <div>
          <h1>Image Gallery</h1>
          <FormContainer addImage={this.addImage} />
          <DisplaySelector setDisplay={this.setDisplay} />
          <DisplayType images={this.state.images} deleteImage={this.deleteImage} />
        </div>
    );
  }

  componentDidMount() {
    fetch('http://localhost:3001/images')
      .then(res => res.json())
      .then(resImages => {
        this.setState({ images: resImages });
      });
  }
  
}

