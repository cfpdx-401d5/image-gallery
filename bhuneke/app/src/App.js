import React, { Component } from 'react';
import AddResource from './components/editing/AddResource';
import GalleryDisplay from './components/display/Gallery';
import ThumbnailDisplay from './components/display/Thumbnail';
import DetailDisplay from './components/display/Detail';
import fetcher from './helpers/fetcher';

const GALLERY_TYPE = 'gallery';
const LIST_TYPE = 'list';
const THUMBNAIL_TYPE = 'thumbnail';

export default class App extends Component {

    constructor(props) {
        super(props);
        this.state = {
            selectedView: '',
            resources: []
        };
        this.onViewSelect = this.onViewSelect.bind(this);
        this.onDelete = this.onDelete.bind(this);
        this.onAdd = this.onAdd.bind(this);
    }

    onViewSelect(e, selected) {
        e.preventDefault();
        this.setState({
            selectedView: selected
        });
    }
    doFetch() {
        fetcher({
            path: '/resources',
            method: 'GET',
        })
        .then(res => res.json())
        .then(resources => {
            this.setState({ resources });
        })
        .catch(err => { console.log(err); });
    }

    componentDidMount() {
        this.doFetch();
    }

    onDelete(id) {
        fetcher({
            path: `/resources/${id}`,
            method: 'DELETE'
        });
        this.doFetch();
    }

    onAdd(newResource) {
        let newData = this.state.resources.slice();
        newData.push(newResource);
        this.setState({
            resources: newData
        });
    }

    render() {
        return (
            <div>
                < ViewSelector onViewSelect={this.onViewSelect} />
                < ViewDisplay onDelete={this.onDelete} selectedView={this.state.selectedView} resources={this.state.resources} />
                < AddResource onAdd={this.onAdd} />
            </div>
        );
    }
}

function ViewSelector(props) {
    return (
        <div className='view-selector'>
            <button onClick={(e, selected) => props.onViewSelect(e, LIST_TYPE)}>List View</button>
            <button onClick={(e, selected) => props.onViewSelect(e, THUMBNAIL_TYPE)}>Thumbnail View</button>
            <button onClick={(e, selected) => props.onViewSelect(e, GALLERY_TYPE)}>Gallery View</button>
        </div>);
}

function ViewDisplay(props) {
    let selectedView = props.selectedView;
    if (selectedView) {
        return (
            <ListView
                selectedView={selectedView}
                resources={props.resources}
                onDelete={props.onDelete} 
            />);
    } else {
        return <p>Choose View!</p>;
    }
}

function ListView(props) {
    let DisplayType;
    const selectedView = props.selectedView;
    if (selectedView === LIST_TYPE) DisplayType = DetailDisplay;
    else if (selectedView === GALLERY_TYPE) DisplayType = GalleryDisplay;
    else if (selectedView === THUMBNAIL_TYPE) DisplayType = ThumbnailDisplay;

    const resourceList = props.resources.map(resource => {
        return (
            <DisplayType
                key={resource._id}
                resource={resource}
                onDelete={props.onDelete} 
            />);
    });

    return (
        <div>{resourceList}</div>
    );
}

export {
    ViewSelector,
    ViewDisplay,
    ListView,
};
