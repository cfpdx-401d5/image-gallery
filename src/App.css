.App {
  text-align: center;
}


.App-header {
  background-color: #222;
  color: white;
}

div #root {
  text-align: center;
  width: 90%%;
  margin-left: auto;
  margin-right: auto;
  border: .5em solid #bc640b;
}

.viewer {
height: 80%;
border: .5em solid #000;
}

img {
  max-height: 100%;
  max-width: 100%;
  padding: 1em;
}