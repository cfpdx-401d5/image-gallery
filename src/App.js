import React, { Component } from 'react';
import AddResource from './add-resource'

const GALLERY_TYPE = 'gallery';
const LIST_TYPE = 'list';
const THUMBNAIL_TYPE = 'thumbnail';

/////////   APP CLASS   /////////
export default class App extends Component {

  constructor(props) {
    super(props);
    this.state = {
      selectedView: '',
      resources: props.resources
    };
    this.onViewSelect = this.onViewSelect.bind(this);
    this.onDelete = this.onDelete.bind(this);
    this.onAdd = this.onAdd.bind(this);
  }

  onViewSelect(e, selected) {
    e.preventDefault();
    
    this.setState({
      selectedView: selected
    })
  }

  onDelete(id) {
      let currentResources = [...this.state.resources];

      currentResources = currentResources.filter(resource => {
          return resource.id !== id;
      })

      this.setState({
          resources: currentResources
      })
  }

  onAdd(e, newResource) {
      e.preventDefault();

      let newData = this.state.resources.slice();
      newData.push(newResource);
      this.setState({
          resources: newData
      })
  }

  render() {
    return (
      <div>
        < ViewSelector onViewSelect={this.onViewSelect}/>
        < ViewDisplay onDelete={this.onDelete} selectedView={this.state.selectedView} resources={this.state.resources}/>
        < AddResource onAdd={this.onAdd}/>
      </div>
    )
  }
}
/////////   END APP CLASS   /////////


function ViewSelector(props) {
  return (
    <div className='view-selector'>
      <button onClick={(e, selected) => props.onViewSelect(e, LIST_TYPE)}>List View</button>
      <button onClick={(e, selected) => props.onViewSelect(e, THUMBNAIL_TYPE)}>Thumbnail View</button>
      <button onClick={(e, selected) =>props.onViewSelect(e, GALLERY_TYPE)}>Gallery View</button>
    </div>)
}

function ViewDisplay(props) {
    let selectedView = props.selectedView;
    if (selectedView === LIST_TYPE) {
        return <DetailList resources={props.resources} onDelete={props.onDelete}/>
    }
    else if (selectedView === THUMBNAIL_TYPE) {
        return <ThumbnailList resources={props.resources} onDelete={props.onDelete}/>
    }
    else if (selectedView === GALLERY_TYPE) {
        return <GalleryList resources={props.resources} onDelete={props.onDelete}/>
    }
    else {
        return <p>Choose View!</p>
    }
}

/////////   LIST COMPONENTS   /////////
function DetailList(props) {
    const detailList = props.resources.map(resource => {
        return <DetailDisplay key={resource.id} resource={resource} onDelete={props.onDelete}/>
    })
    return (
        <div>{detailList}</div>
    );
}

function ThumbnailList(props) {
    const thumbnailList = props.resources.map(resource => {
        return <ThumbnailDisplay key={resource.id} resource={resource} onDelete={props.onDelete}/>
    })
    return (
        <div>{thumbnailList}</div>
    );
}

function GalleryList(props) {
    const galleryList = props.resources.map(resource => {
        return <GalleryDisplay  key={resource.id} resource={resource} onDelete={props.onDelete}/>
    })
    return (
        <div>{galleryList}</div>
    );
}
/////////   END LIST COMPONENTS   /////////

/////////   DISPLAY COMPONENTS   /////////
function DetailDisplay(props) {
  return (
    <div>
        <ul>
        <li>Title: {props.resource.title}</li>
        <li>Url: {props.resource.url}</li>
        <li>Description: {props.resource.description}</li>
        </ul>
        <button onClick={() => props.onDelete(props.resource.id)}>delete</button>
    </div>
  )  
}

function ThumbnailDisplay(props) {
  return (
    <div>
      <p>Title: {props.resource.title}</p>
      <img className='thumbnail' src={props.resource.url} alt='Cute Bunny'/>
      <button onClick={() => props.onDelete(props.resource.id)}>delete</button>
    </div>
  )    
}

function GalleryDisplay(props) {
  return (
    <div>
      <p>Title: {props.resource.title}</p>
      <img className='gallery' src={props.resource.url} alt='Cute Bunny'/>
      <p>Description: {props.resource.description}</p>
      <button onClick={() => props.onDelete(props.resource.id)}>delete</button>
    </div>
  ) 
}
/////////   END DISPLAY COMPONENTS   /////////


/////////   EXPORT   /////////
export {
    ViewSelector,
    ViewDisplay,
    DetailList,
    ThumbnailList,
    GalleryList,
    DetailDisplay,
    ThumbnailDisplay,
    GalleryDisplay
}