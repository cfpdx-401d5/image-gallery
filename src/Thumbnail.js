import React, { PropTypes } from 'react';

function Thumbnail(props) {
  let altText = props.title + ' thumbnail';

  return (
    <figure>
      <img className='thumbnail' src={props.url} alt={altText} />
      <figcaption>{props.title}</figcaption>
    </figure>
  );
}
Thumbnail.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default Thumbnail;
