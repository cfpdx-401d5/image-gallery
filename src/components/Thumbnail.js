import React from 'react';

export default function Thumbnail(props) {
    return (
        <div>
            <img alt={props.image.title} src={props.image.url} height='100px' width='100px' />
            <h4>{props.image.title}</h4>
        </div>
    );
}