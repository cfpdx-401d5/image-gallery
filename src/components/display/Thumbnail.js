import React from 'react';

export default function ThumbnailDisplay(props) {
    return (
      <div>
        <h1>Title: {props.resource.title}</h1>
        <img className='thumbnail' src={props.resource.url} alt='Cute Bunny' />
        <button onClick={() => props.onDelete(props.resource.id)}>delete</button>
      </div>
    );
};
