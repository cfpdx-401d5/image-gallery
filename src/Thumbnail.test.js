import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './App';
import Thumbnail from './Thumbnail';
let image = {
    title: 'Cute Bunny',
    description: 'Isn\'t it fuzzy-wuzzy cutest thing you\'ve ever seen?',
    url: 'http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg'
};


describe('Thumbnail', () => {
    it('runs the Thumbnail component', () => {
        const rendered = shallow(<Thumbnail image={image} />);
        const json = toJson(rendered);
        expect(json).toMatchSnapshot();
    })

});