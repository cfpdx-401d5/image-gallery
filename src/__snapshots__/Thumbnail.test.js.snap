exports[`Thumbnail runs the Thumbnail component 1`] = `
<div>
  <img
    alt="Cute Bunny"
    height="100px"
    src="http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg"
    width="100px" />
  <h4>
    Cute Bunny
  </h4>
</div>
`;
